### PR TITLE
Test encrypted+multipart download at remote site(active/archive)

### PR DESCRIPTION
**Pull request by viduship**
Test encrypted+multipart download at remote site(active/archive)

Jira ticket : https://issues.redhat.com/browse/RHCEPHQE-12557

Polarian:

1. https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-11357
2. https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83573390

Bugs:

1. https://bugzilla.redhat.com/show_bug.cgi?id=2237838
2. https://bugzilla.redhat.com/show_bug.cgi?id=2162337

Passed logs: 

http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-RKP9JO

cephci PR : https://github.com/red-hat-storage/cephci/pull/3289

### CODE REVIEW

File: rgw/v2/tests/s3_swift/multisite_configs/test_sse_s3_bucket_enc_multipart_download_remote_site.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#polarian CEPH-11357, bug-2162337
+#script- test_sse_s3_kms_with_vault.py
+config:
+ user_count: 1
+ encryption_keys: s3
+ bucket_count: 1
```
Comment (mkasturi18) [inline rgw/v2/tests/s3_swift/multisite_configs/test_sse_s3_bucket_enc_multipart_download_remote_site.yaml:6]: @viduship , do we need "remote_zone: archive" here as well

File: rgw/v2/tests/s3_swift/reusable.py [hunk 2]
```diff
@@ -2099,22 +2099,39 @@ def test_object_download_at_replicated_site(
...
         remote_site_auth = Auth(
             each_user, remote_site_ssh_conn, ssl=config.ssl, haproxy=config.haproxy
         )
         remote_s3_client = remote_site_auth.do_auth_using_client()
-
-        log.info(
-            f"Download object {s3_object_name} via boto3 at remote site {zone_name}"
+        if zone_name == "archive":
+            log.info(f"It is a {zone_name} zone, hence objects are always versioned.")
+        time.sleep(20)
```
Comment (mkasturi18) [inline rgw/v2/tests/s3_swift/reusable.py:2116]: Any reason to induce delay here? we are already wait for sync in #L209

**Review by mkasturi18 — CHANGES_REQUESTED** [review]

File: rgw/v2/tests/s3_swift/multisite_configs/test_sse_s3_bucket_enc_multipart_download_remote_site.yaml (hunk 1 above)
Comment (viduship) [reply rgw/v2/tests/s3_swift/multisite_configs/test_sse_s3_bucket_enc_multipart_download_remote_site.yaml:6]: @mkasturi18 , we do not. For active zones it picks up the remote site based on the previous methods already in ceph-qe-scripts.

File: rgw/v2/tests/s3_swift/reusable.py (hunk 2 above)
Comment (viduship) [reply rgw/v2/tests/s3_swift/reusable.py:2116]: @mkasturi18, we don't get the 'versions' in the response details of the object.

**Review by mkasturi18 — COMMENTED** [review]
Changes looks good to me

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**